crates/mockiato-codegen/src/generate/visibility.rs

```diff
@@ -0,0 +1,44 @@
+use std::iter;
+use syn::{parse_quote, PathSegment, VisRestricted, Visibility};
+
+pub(super) fn raise_visibility_by_one_level(visibility: &Visibility) -> Visibility {
+    match visibility {
+        Visibility::Inherited => parse_quote!(pub(super)),
+        Visibility::Public(_) | Visibility::Crate(_) => visibility.clone(),
+        Visibility::Restricted(restricted) if has_leading_colon(restricted) => visibility.clone(),
```

Comment:
Maybe this would be more readable if the `if` expression was in the match instead of a guard. Then, the `Visibility::Restricted(restricted)` variant would not be matched twice

Reply:
Resolves cf48747